http_queue_service.go

```diff
@@ -0,0 +1,59 @@
+package main
+
+import (
+	"bytes"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"strings"
+	"time"
+
+	log "github.com/Sirupsen/logrus"
+)
+
+type httpClient interface {
+	Do(req *http.Request) (resp *http.Response, err error)
+}
+
+type httpQueue struct {
+	httpClient httpClient
+	endpoint   string
+}
+
+func newHttpQueueService(httpClient httpClient, endpoint string) httpQueue {
+	return httpQueue{
+		httpClient: httpClient,
+		endpoint:   endpoint,
+	}
+}
+
+func (q httpQueue) sendMessage(id string, conceptType string, tid string, payload []byte) error {
+
+	url, err := url.Parse(strings.TrimRight(q.endpoint, "/") + "/" + id)
+	if err != nil {
+		log.Errorf("Error generating URL: %s", err)
+		return err
+	}
+	log.Debug(url)
+	req := &http.Request{
+		Method: "PUT",
+		URL:    url,
+		Header: http.Header{
+			"Message-Type":      {conceptType},
+			"Content-Type":      {"application/json"},
+			"X-Request-Id":      {tid},
+			"Origin-System-Id":  {"http://cmdb.ft.com/systems/upp"},
```

Comment:
I suggest that we replace "X-Request-Id" literal with the common library constants github.com/Financial-Times/transactionid-utils-go.  It is there to do transaction_id (algo and names) uniformly across all services. Also it would be nice to use the library code rather than generateID() in publish_service.go L71 & L290.  We should avoid literals as it is bad coding practice.